### PR TITLE
✨ Initial e2e test

### PR DESCRIPTION
**Pull request by waltforme**
## Summary
This PR introduces initial e2e test.

The test currently has one simple scenario, which is creating a ManifestWork (for a Deployment) then check the creation and content of the corresponding WorkStatus.

The test has been exercised on linux/arm64 and darwin/arm64.

On darwin/arm64, `make manifests` somehow fails. This is handled in this PR by updating the version of controller-gen.

## Related issue(s)

#26

### CODE REVIEW

**Comment by andreyod** [comment]
Maybe add to the README a short prerequisites section and the test description.

File: test/e2e/workstatus-creation.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+set -x # echo so that users can understand what is happening
+set -e # exit on error
+
+:
+: -------------------------------------------------------------------------
+: Build and deploy the addon
+:
+make ko-local-build
+make kind-load-image CLUSTERS="hub cluster1"
```
Comment (ezrasilvera) [inline test/e2e/workstatus-creation.sh:24]: I know it's not part of the PR, but I really prefer we would load images with a command or a script and not through `make`  target

**Comment by ezrasilvera** [comment]
@waltforme Do you also want to add a workflow that will trigger this on PRs?  it's very easy you can copy `https://github.com/kubestellar/kubestellar/blob/main/.github/workflows/pr-test-multicluster.yml`    and change it .

File: .github/workflows/workstatus-creation.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  workflow_dispatch:
+  # To confirm any changes to docs build successfully, without deploying them
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
```
[the remark this answers is not in the document]
Comment (waltforme) [reply .github/workflows/workstatus-creation.yaml:15]: I guess PRs are prescribed in line 10? Not an export of GH Actions though.

**Comment by waltforme** [comment]
> Maybe add to the README a short prerequisites section and the test description.

Thanks! Added as commit https://github.com/kubestellar/ocm-status-addon/pull/27/commits/3b983a29e96614028ea3b9d00ea9ea04fde0e25b.

File: test/e2e/workstatus-creation.sh (hunk 1 above)
Comment (ezrasilvera) [reply test/e2e/workstatus-creation.sh:24]: This is general for all our Makefiles. So let's close it here, and open as general issue .

**Review by ezrasilvera — APPROVED** [review]

**Pull request merged**